File: python/test/test_find.py
import ThrustRTC as trtc



d_values = trtc.device_vector_from_list([0, 5, 3, 7], 'int32_t')
print(trtc.Find(d_values, trtc.DVInt32(3)))
print(trtc.Find(d_values,trtc.DVInt32(5)))
print(trtc.Find(d_values,trtc.DVInt32(9)))
print(trtc.Find_If(d_values, trtc.Functor({}, ['x'], '        return x>4;\n')))
print(trtc.Find_If(d_values, trtc.Functor({}, ['x'], '        return x>10;\n')))
print(trtc.Find_If_Not(d_values, trtc.Functor({}, ['x'], '        return x>4;\n')))
print(trtc.Find_If_Not(d_values, trtc.Functor({}, ['x'], '        return x>10;\n')))

File: java/test/test_find.java
import java.util.*;
import JThrustRTC.*;

public class test_find
{
	public static void main(String[] args) 
	{
		DVVector d_values = new DVVector(new int[] { 0, 5, 3, 7 });
		System.out.println(TRTC.Find(d_values, new DVInt32(3)));
		System.out.println(TRTC.Find(d_values, new DVInt32(5)));
		System.out.println(TRTC.Find(d_values, new DVInt32(9)));
	}
}